src/components/calcite-input/calcite-input.tsx

```diff
@@ -431,7 +431,8 @@ export class CalciteInput {
     if (this.childElType === "input" && this.type === "number") {
       const inputMax = this.maxString ? parseFloat(this.maxString) : null;
       const inputMin = this.minString ? parseFloat(this.minString) : null;
-      const inputStep = this.stepString ? parseFloat(this.stepString) : 1;
+      const inputStep =
```

Comment:
Nitpick: `const inputStep = this.stepString > 0 ? parseFloat(this.stepString) : 1;`

Reply:
The ts compiler does not like using > with a string. I would have to change this to using the `step` prop directly instead of `stepString` maybe?

Reply:
Ah, good point. I'm not sure about the using the `step` prop directly, but I think this just needs a hint that it's dealing with a number type:

```ts
const inputStep = 
  typeof this.stepString === "number" && this.stepString > 0 ? 
  parseFloat(this.stepString) : 1;
```

Reply:
The first condition is always false I think, how about:
```ts
const inputStep = Number(this.stepString) > 0 ? parseFloat(this.stepString) : 1;
```

Reply:
You are correct. That'll do it. 

Coffee hadn't kicked in, I wasn't paying attention to the `String` in the prop name. 😅